pkg/pool-utils/contracts/RecoveryModeHelper.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/solidity-utils/helpers/BalancerErrors.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/BasePoolUserData.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IRecoveryModeHelper.sol";
+
+import "@balancer-labs/v2-pool-weighted/contracts/WeightedMath.sol";
+
+import "./lib/ComposablePoolLib.sol";
+
+contract RecoveryModeHelper is IRecoveryModeHelper {
+    using BasePoolUserData for bytes;
+
+    IVault private immutable _vault;
+
+    constructor(IVault vault) {
+        _vault = vault;
+    }
+
+    function getVault() public view returns (IVault) {
+        return _vault;
+    }
+
+    function calcComposableRecoveryAmountsOut(
+        bytes32 poolId,
+        bytes memory userData,
+        uint256 totalSupply
+    ) external view override returns (uint256 bptAmountIn, uint256[] memory amountsOut) {
+        // As ManagedPool is a composable Pool, `_doRecoveryModeExit()` must use the virtual supply rather than the
```

Comment:
```suggestion
        // As this is a composable Pool, `_doRecoveryModeExit()` must use the virtual supply rather than the
```
If the intent is to use this for all kinds of composable pools (linear, stable as well), we should avoid mentions of explicit pool types

Reply:
Ah yes, i forgot to change this. Thanks!